pkg/oci/load_balancer.go

```diff
@@ -180,10 +186,10 @@ func getSubnetsForNodes(ctx context.Context, nodes []*v1.Node, client client.Int
 
 // readSSLSecret returns the certificate and private key from a Kubernetes TLS
 // private key Secret.
-func (cp *CloudProvider) readSSLSecret(svc *v1.Service) (string, string, error) {
-	secretString, ok := svc.Annotations[ServiceAnnotationLoadBalancerTLSSecret]
-	if !ok {
-		return "", "", errors.Errorf("no %q annotation found", ServiceAnnotationLoadBalancerTLSSecret)
+func (cp *CloudProvider) readSSLSecret(secretType string, svc *v1.Service) (string, string, string, string, error) {
```

Comment:
The return type here is pretty hard to understand and might be prone to error. Suggest refactoring into a struct or similar.

Reply:
You're right, thanks!